### ISSUE TITLE
Automated Payload Injection Post-Jailbreak for Linux Boot

### ISSUE THREAD

**Issue by seuros** [xfangfang/PPPwn_cpp#82]
I've got an idea that could really improve the experience for those of us who use the PS4 mostly for its Linux capabilities. Here's what I'm thinking:

Right now, after we jailbreak the PS4, we have to manually inject additional payloads to get into Linux. It would be awesome if we could modify this app process so that it automatically injects these payloads once the jailbreak is complete.

My idea is to have a folder named **payloads** where we can store all the necessary payloads. The system would then automatically send these payloads in alphabetical order right after the jailbreak finishes. 
This way, I could just leave my console on PPP all the time, and it would autoboot into Linux without any extra steps.

This would make the whole process much smoother and more convenient, especially for those of us who use Linux on the PS4 regularly. No more fiddling around with payloads every time we reboot – just jailbreak and go straight into Linux.

What do you think? Would anyone else find this useful?

I could try to give it a stab if there is demand. 
Right now i just have script that automated payload after this application exit.

**Comment by Gouster4** [comment]
I think that's kinda impossible, because we are limited to payload size. That's why GoldHen is loaded first time from USB stick, then from USB. It's done by pppw 2nd stage shipped by GoldHen. It is possible to load all payloads to USB stick, and make own 2nd stage that loads them from USB or even copy to HDD and then loads from there like GoldHen does. But it's not related to PPPwn itself, as PPPwn is just tool to load that 2nd stage.

Requires new project that aims to do this. Take a look on GoldHen source code, maybe you can modify theyr 2nd stage for your needs.

**Comment by seuros** [comment]
After injecting stage2, goldenhen is started and the binloader server start listening with the fake ip 42.42.42.42 for  about 40 seconds before shut-downing.

If we could push valid payloads in this time period, we could  boot linux without user intervention.

**Comment by Gouster4** [comment]
> After injecting stage2, goldenhen is started and the binloader server start listening with the fake ip 42.42.42.42 for about 40 seconds before shut-downing.
> 
> If we could push valid payloads in this time period, we could boot linux without user intervention.

To get internet access on my PS4, after sucesfull pppwn, I'm starting regular pppoe server, where PS4 gets IP I set before, so that is also an option for binloader.

**Comment by seuros** [comment]
It is an automated process or you have to do it manually ?

**Comment by Gouster4** [comment]
> It is an automated process or you have to do it manually ?

Manually, using mpd5, or automatically with own script I made.

But you can use gateway branch of pppwn to get internet connection on PS4 after jailbreak.

**Comment by kappafon** [comment]
@Gouster4 could you please share the script you mentioned?  I also have opnSense but still kind of new to it's capabilities.
Do you have any useful links or steps to share on setting up pppoe server? Was researching a bit a while back but then postponed it.

I managed to build pppwn_cpp for opnSense and atm I ssh into opnSense every time I boot up my PS4
Then paste the command to run pppwn. Also have to Test internet connection on PS4 or it won't start the actual process.

Then when I want to transfer something I plug the cable back into my LAN and re-setup internet connection on PS4.

Any help is appreciated :)
